src/language/online-parser/parser.js

```diff
@@ -0,0 +1,650 @@
+import { Lexer } from '../lexer';
+import { Source } from '../source';
+
+import Language from './language';
+import { RuleKind, TokenKind } from './types';
+import type {
+  Token,
+  LexerToken,
+  Styles,
+  ParserRule,
+  ParserState,
+  ParserConfig,
+  ParserConfigOption,
+  Rule,
+  RuleName,
+  RuleConstraint,
+  TokenParserRule,
+  OfTypeParserRule,
+  ListOfTypeParserRule,
+  PeekParserRule,
+  ConstraintsSetRule,
+  OfTypeConstraint,
+  ListOfTypeConstraint,
+  TokenConstraint,
+  PeekConstraint,
+  ConstraintsSet,
+} from './types';
+
+export class Parser {
+  state: ParserState;
+  lexer: Lexer;
+  styles: Styles;
+  config: ParserConfig;
+
+  constructor({
+    state,
+    styles,
+    config,
+    source,
+  }: {|
```

Comment:
Can you please convert to positional arguments?
especially since we need to remove styles.